Sources/MongoSwift/BSON/Document.swift

```diff
@@ -323,6 +324,22 @@ extension Document {
         self._storage = DocumentStorage()
     }
 
+    internal init(keyValuePairs: [(String, BSON)]) {
```

Comment:
this was previously just contained in `Document`'s `ExpressibleByDictionaryLiteral` conformance, but I needed to move it here so that `BSON` could use it in its own `ExpressibleByDictionaryLiteral` conformance.